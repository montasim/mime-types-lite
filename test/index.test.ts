import mimeTypesLite, { MimeType } from '../src/index';

describe('mimeTypesLite', () => {
    /**
     * Ensures the MIME types module is defined and immutable.
     */
    it('should be defined and frozen', () => {
        expect(mimeTypesLite).toBeDefined();
        expect(Object.isFrozen(mimeTypesLite)).toBe(true); // Ensures the object is immutable
    });

    /**
     * Verifies correct MIME types for document-related keys.
     */
    it('should return correct MIME types for documents', () => {
        expect(mimeTypesLite.PDF).toBe('application/pdf');
        expect(mimeTypesLite.MD).toBe('text/markdown');
        expect(mimeTypesLite.TXT).toBe('text/plain');
        expect(mimeTypesLite.CSV).toBe('text/csv');
        expect(mimeTypesLite.DOC).toBe('application/msword');
        expect(mimeTypesLite.DOCX).toBe(
            'application/vnd.openxmlformats-officedocument.wordprocessingml.document'
        );
        expect(mimeTypesLite.XLS).toBe('application/vnd.ms-excel');
        expect(mimeTypesLite.XLSX).toBe(
            'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet'
        );
        expect(mimeTypesLite.PPT).toBe('application/vnd.ms-powerpoint');
        expect(mimeTypesLite.PPTX).toBe(
            'application/vnd.openxmlformats-officedocument.presentationml.presentation'
        );
        expect(mimeTypesLite.RTF).toBe('application/rtf');
        expect(mimeTypesLite.EPUB).toBe('application/epub+zip');
        expect(mimeTypesLite.TEX).toBe('application/x-tex');
        expect(mimeTypesLite.ODS).toBe(
            'application/vnd.oasis.opendocument.spreadsheet'
        );
        expect(mimeTypesLite.ODT).toBe(
            'application/vnd.oasis.opendocument.text'
        );
    });

    /**
     * Verifies correct MIME types for image-related keys.
     */
    it('should return correct MIME types for images', () => {
        expect(mimeTypesLite.JPG).toBe('image/jpeg');
        expect(mimeTypesLite.JPEG).toBe('image/jpeg');
        expect(mimeTypesLite.PNG).toBe('image/png');
        expect(mimeTypesLite.ICO).toBe('image/x-icon');
        expect(mimeTypesLite.GIF).toBe('image/gif');
        expect(mimeTypesLite.BMP).toBe('image/bmp');
        expect(mimeTypesLite.TIFF).toBe('image/tiff');
        expect(mimeTypesLite.SVG).toBe('image/svg+xml');
        expect(mimeTypesLite.WEBP).toBe('image/webp');
        expect(mimeTypesLite.HEIC).toBe('image/heic');
        expect(mimeTypesLite.AVIF).toBe('image/avif');
        expect(mimeTypesLite.JP2).toBe('image/jp2');
        expect(mimeTypesLite.PSD).toBe('image/vnd.adobe.photoshop');
        expect(mimeTypesLite.XCF).toBe('image/x-xcf');
    });

    /**
     * Verifies correct MIME types for video-related keys.
     */
    it('should return correct MIME types for videos', () => {
        expect(mimeTypesLite.MP4).toBe('video/mp4');
        expect(mimeTypesLite.WEBM).toBe('video/webm');
        expect(mimeTypesLite.AVI).toBe('video/avi');
        expect(mimeTypesLite.MPEG).toBe('video/mpeg');
        expect(mimeTypesLite.MKV).toBe('video/x-matroska');
        expect(mimeTypesLite.MOV).toBe('video/quicktime');
        expect(mimeTypesLite.FLV).toBe('video/x-flv');
        expect(mimeTypesLite.WMV).toBe('video/x-ms-wmv');
    });

    /**
     * Verifies correct MIME types for audio-related keys.
     */
    it('should return correct MIME types for audio files', () => {
        expect(mimeTypesLite.MP3).toBe('audio/mpeg');
        expect(mimeTypesLite.WAV).toBe('audio/wav');
        expect(mimeTypesLite.AAC).toBe('audio/aac');
        expect(mimeTypesLite.OGG).toBe('audio/ogg');
        expect(mimeTypesLite.FLAC).toBe('audio/flac');
        expect(mimeTypesLite.MIDI).toBe('audio/midi');
        expect(mimeTypesLite.AMR).toBe('audio/amr');
    });

    /**
     * Verifies correct MIME types for archive-related keys.
     */
    it('should return correct MIME types for archives', () => {
        expect(mimeTypesLite.ZIP).toBe('application/zip');
        expect(mimeTypesLite.RAR).toBe('application/vnd.rar');
        expect(mimeTypesLite.TAR).toBe('application/x-tar');
        expect(mimeTypesLite.GZ).toBe('application/gzip');
        expect(mimeTypesLite.BZ2).toBe('application/x-bzip2');
        expect(mimeTypesLite.SEVEN_ZIP).toBe('application/x-7z-compressed');
    });

    /**
     * Verifies correct MIME types for web-related keys.
     */
    it('should return correct MIME types for web-related files', () => {
        expect(mimeTypesLite.JSON).toBe('application/json');
        expect(mimeTypesLite.XML).toBe('application/xml');
        expect(mimeTypesLite.JS).toBe('application/javascript');
        expect(mimeTypesLite.CSS).toBe('text/css');
        expect(mimeTypesLite.HTML).toBe('text/html');
        expect(mimeTypesLite.WASM).toBe('application/wasm');
        expect(mimeTypesLite.YAML).toBe('application/x-yaml');
        expect(mimeTypesLite.GRAPHQL).toBe('application/graphql');
        expect(mimeTypesLite.ICS).toBe('text/calendar');
        expect(mimeTypesLite.ATOM).toBe('application/atom+xml');
        expect(mimeTypesLite.RSS).toBe('application/rss+xml');
    });

    /**
     * Verifies correct MIME types for font-related keys.
     */
    it('should return correct MIME types for fonts', () => {
        expect(mimeTypesLite.WOFF).toBe('font/woff');
        expect(mimeTypesLite.WOFF2).toBe('font/woff2');
        expect(mimeTypesLite.TTF).toBe('font/ttf');
        expect(mimeTypesLite.OTF).toBe('font/otf');
    });

    /**
     * Validates that only valid keys are allowed as MimeType type.
     * Ensures TypeScript type safety for valid and invalid keys.
     */
    it('should allow only valid keys as MimeType type', () => {
        const validMimeType: MimeType = 'PDF';
        expect(mimeTypesLite[validMimeType]).toBe('application/pdf');

        // @ts-expect-error This should throw a TypeScript error during compilation
        const invalidMimeType: MimeType = 'INVALID';
        // Uncomment below to test runtime behavior if needed:
        expect(mimeTypesLite[invalidMimeType]).toBeUndefined();
    });

    /**
     * Creates a snapshot of the entire MIME types object for consistency checks.
     */
    it('should match snapshot for the entire constants object', () => {
        expect(mimeTypesLite).toMatchSnapshot();
    });
});
