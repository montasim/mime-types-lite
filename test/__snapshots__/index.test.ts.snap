// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`mimeTypesLite should match snapshot for the entire constants object 1`] = `
{
  "AAC": "audio/aac",
  "AMR": "audio/amr",
  "ATOM": "application/atom+xml",
  "AVI": "video/avi",
  "AVIF": "image/avif",
  "BMP": "image/bmp",
  "BZ2": "application/x-bzip2",
  "CSS": "text/css",
  "CSV": "text/csv",
  "CSV_SEMICOLON": "text/csv+semicolon",
  "DOC": "application/msword",
  "DOCX": "application/vnd.openxmlformats-officedocument.wordprocessingml.document",
  "EPUB": "application/epub+zip",
  "FLAC": "audio/flac",
  "FLV": "video/x-flv",
  "GIF": "image/gif",
  "GRAPHQL": "application/graphql",
  "GZ": "application/gzip",
  "HEIC": "image/heic",
  "HTML": "text/html",
  "ICO": "image/x-icon",
  "ICS": "text/calendar",
  "JP2": "image/jp2",
  "JPEG": "image/jpeg",
  "JPG": "image/jpeg",
  "JS": "application/javascript",
  "JSON": "application/json",
  "MD": "text/markdown",
  "MIDI": "audio/midi",
  "MKV": "video/x-matroska",
  "MOV": "video/quicktime",
  "MP3": "audio/mpeg",
  "MP4": "video/mp4",
  "MPEG": "video/mpeg",
  "ODS": "application/vnd.oasis.opendocument.spreadsheet",
  "ODT": "application/vnd.oasis.opendocument.text",
  "OGG": "audio/ogg",
  "OTF": "font/otf",
  "PDF": "application/pdf",
  "PNG": "image/png",
  "PPT": "application/vnd.ms-powerpoint",
  "PPTX": "application/vnd.openxmlformats-officedocument.presentationml.presentation",
  "PSD": "image/vnd.adobe.photoshop",
  "RAR": "application/vnd.rar",
  "RSS": "application/rss+xml",
  "RTF": "application/rtf",
  "SEVEN_ZIP": "application/x-7z-compressed",
  "SVG": "image/svg+xml",
  "TAR": "application/x-tar",
  "TEX": "application/x-tex",
  "TIFF": "image/tiff",
  "TTF": "font/ttf",
  "TXT": "text/plain",
  "URL_ENCODED": "application/x-www-form-urlencoded",
  "WASM": "application/wasm",
  "WAV": "audio/wav",
  "WEBM": "video/webm",
  "WEBP": "image/webp",
  "WMV": "video/x-ms-wmv",
  "WOFF": "font/woff",
  "WOFF2": "font/woff2",
  "XCF": "image/x-xcf",
  "XLS": "application/vnd.ms-excel",
  "XLSX": "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
  "XML": "application/xml",
  "YAML": "application/x-yaml",
  "ZIP": "application/zip",
}
`;
